docs/contributor/test-plan-unittesting.md

```diff
@@ -0,0 +1,128 @@
+# Unittesting Test Plan
+
+## **Purpose** 
+
+The unit testing phase focuses on maintainability of the code. This involves testing the logic of smallest grains of the code in isolation and it aims to prevent regression resulted by code changes. 
+
+## Scope
+
+Unit test suits are the lowest-level test suites, and they test the smallest grains of logic, mainly methods, in isolation, to prevent regression resulted by code changes. Unit tests should be able to run in isolation independent of external systems such as databases, external APIs, etc. 
```

Comment:
```suggestion
Unit test suits are the lowest-level test suites, and they test the smallest grains of logic, mainly methods, in isolation, to prevent regression resulting from code changes. Unit tests should be able to run in isolation independent of external systems such as databases, external APIs, etc. 
```